### ISSUE TITLE
Clarification Needed on Few-Shot Experiment Settings for 16 Shots

### ISSUE THREAD

**Issue by ChushanZhang** [muzairkhattak/PromptSRC#13]
Hello there @muzairkhattak,

Thank you for this work!

I am running the few-shot experiments, but I am a bit confused about the 16-shot setting. I checked the base-to-novel script, and its default is shot=16. However, for the few-shot benchmark, I noticed that the results for 16 shots are different from the base-to-novel.

Is there anything I am missing? Besides that, for the result obtained(e.g.1 shot for Imagenet 68.13), is that refers to novel(test) only or training eval or HM score? 

I appreciate your help in advance!

**Comment by ChushanZhang** [comment]
And there is a minor qeustion hope you can help with: I found you did few-shots experiments on MaPLe as well, but I cannot find the config file for that. Could you kindly share the setting with me?

**Comment by muzairkhattak** [comment]
Hi @PoopBear1, thank you for your message.

> I am running the few-shot experiments, but I am a bit confused about the 16-shot setting. I checked the base-to-novel script, and its default is shot=16. However, for the few-shot benchmark, I noticed that the results for 16 shots are different from the base-to-novel.

For base-to-novel settings, although we use 16 shots, models are trained only on the base classes while evaluation is done on both base and novel classes. For few-shot experiments, we train the model on all classes, due to which the results for 16 shots is different for base2novel and few-shot setting. 

> Besides that, for the result obtained(e.g.1 shot for Imagenet 68.13), is that refers to novel(test) only or training eval or HM score?

All results in the few-shot table refers to the accuracy metric an the models are trained on all classes of the datasets (no splitting of classes occurs in few-shot setting)

> And there is a minor qeustion hope you can help with: I found you did few-shots experiments on MaPLe as well, but I cannot find the config file for that. Could you kindly share the setting with me?

I think we do not have the exact config files for MaPLe for few-shot experiments. However MaPLe is trained using its default batch-size, learning rate (as of its basetonovel setting) and training is carried out for 50 epochs. 

I hope that is helpful
Thank you and kind regards!

**Comment by ChushanZhang** [comment]
Very much appreciated your detailed reply!

**Comment by debarshigit** [comment]
For MaPLe few shot experiments, batch size 4, lr 0.0035 (same as basetonovel setting) and epochs 50 gives different accuracy numbers (average over 3 seeds) than what is reported in the Table. Could you please provide the proper parameters for MaPLe few shot learning to reproduce the results?
